fabric-data-attachment-api-v1/src/main/java/net/fabricmc/fabric/impl/attachment/sync/AttachmentChange.java

```diff
@@ -132,7 +139,26 @@ public Object decodeValue(DynamicRegistryManager dynamicRegistryManager) {
 		return codec.decode(buf);
 	}
 
-	public void apply(World world) {
-		targetInfo.getTarget(world).setAttached((AttachmentType<Object>) type, decodeValue(world.getRegistryManager()));
+	public boolean tryApply(World world, PacketSender sender) {
+		AttachmentTarget target = targetInfo.getTarget(world);
+		Object value = decodeValue(world.getRegistryManager());
+
+		if (target == null) {
+			final MutableText errorMessageText = Text.empty();
+			errorMessageText.append(Text.literal("Received attachment change for unknown target!").formatted(Formatting.RED)).append(ScreenTexts.LINE_BREAK);
+			errorMessageText.append(ScreenTexts.LINE_BREAK);
+
+			errorMessageText.append("Attachment identifier: ").append(Text.literal(String.valueOf(type.identifier())).formatted(Formatting.YELLOW)).append(ScreenTexts.LINE_BREAK);
+			errorMessageText.append("World: ").append(Text.literal(String.valueOf(world.getRegistryKey().getValue())).formatted(Formatting.YELLOW)).append(ScreenTexts.LINE_BREAK);
+			targetInfo.appendDebugInformation(errorMessageText);
+
+			AttachmentEntrypoint.LOGGER.error(errorMessageText.getString());
+			sender.disconnect(errorMessageText);
+
+			return false;
```

Comment:
I think this would be better of throwing a custom exception that passes the Text up to reciever where you can handle the disconnect. (see RemapException as an example)